test/transformers/test_kl_div.py

```diff
@@ -0,0 +1,95 @@
+from test.utils import assert_verbose_allclose, supports_bfloat16
+
+import pytest
+import torch
+from torch.nn import KLDivLoss
+
+from liger_kernel.transformers.kl_div import LigerKLDIVLoss
+
+_SHAPE_PARAMS = (
+    "B, T, V",
+    [
+        (1, 4096, 32000),
+        pytest.param(
+            1,
+            4096,
+            128256,
+            marks=pytest.mark.skipif(
+                torch.cuda.get_device_properties(0).total_memory
+                < 36 * 1000 * 1000 * 1000,
+                reason="This test requires a GPU with at least 36GB of memory",
+            ),
+        ),
+        (3, 423, 32000),
+    ],
+)
+
+_DTYPE_PARAMS = (
+    "dtype, atol, rtol",
+    [
+        pytest.param(
+            torch.bfloat16,
+            1e-8,
+            5e-2,
+            marks=pytest.mark.skipif(
+                not supports_bfloat16(), reason="bfloat16 not supported on this GPU"
+            ),
+        ),
+        (torch.float32, 1e-8, 1e-6),
+        (torch.float16, 1e-3, 1e-2),
+    ],
+)
+
+
+def _test_correctness_once(
+    target_kldiv, B, T, V, dtype, atol, rtol, reduction, log_target, is_last=True
+):
+    torch.manual_seed(0)
+    torch_kldiv = KLDivLoss(reduction=reduction, log_target=log_target)
+
+    input = torch.randn(
+        B * T, V, device="cuda", dtype=dtype, requires_grad=True
```

Comment:
make device as an arg instead of "cuda" since people may want to use it on AMD GPU